packages/ember-metal/lib/computed.js

```diff
@@ -257,6 +257,12 @@ ComputedPropertyPrototype.property = function() {
         `Please refactor from \`Ember.computed('${property}', function() {});\` to \`Ember.computed('${property.slice(0, -6)}.[]', function() {})\`.`,
       property.slice(-5) !== '@each'
     );
+    warn(
+      `Dependent keys containing @each only work one level deep. ` +
+        `You cannot use nested forms like todos.@each.owner.name or todos.@each.owner.@each.name`,
+      /@each(\.[^.]+){2,}/.test(property) === false,
```

Comment:
the capture is not used so this should be a non capturing block, but you could just detect another period after the non period.  `/\.@each\.[^.]+\.` should work.